src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Created.cs

```diff
@@ -111,4 +111,46 @@ public static void FileSystemWatcher_Created_Negative()
             Utility.ExpectNoEvent(eventOccured, "created");
         }
     }
+
+    [Fact]
+    public static void FileSystemWatcher_Created_FileCreatedInNestedDirectory()
+    {
+        TestNestedDirectoriesHelper(WatcherChangeTypes.Created, (AutoResetEvent are, TemporaryTestDirectory ttd) =>
+        {
+            using (var nestedFile = new TemporaryTestFile(Path.Combine(ttd.Path, "nestedFile")))
+            {
+                Utility.ExpectEvent(are, "nested file created", 1000 * 30);
+            }
+        });
+    }
+
+    protected static void TestNestedDirectoriesHelper(
```

Comment:
Since this is used across multiple files, consider putting it into some more central/shared location.